stonesoup/types/particle.py

```diff
@@ -25,3 +29,71 @@ def __init__(self, state_vector, weight, parent=None, *args, **kwargs):
     @property
     def ndim(self):
         return self.state_vector.shape[0]
+
+
+class Particles(Type):
+    """
+    Particle type
+
+    A collection of particles. Contains a state and weight for each particle
+    """
+    state_vector: StateVectors = Property(default=None, doc="State vectors of particles")
+    weight: MutableSequence[Probability] = Property(default=None, doc='Weights of particles')
+    parent: 'Particles' = Property(default=None, doc='Parent particles')
+    particle_list: MutableSequence[Particle] = Property(default=None,
```

Comment:
Why do you need a collection of `StateVector`s and a list of `Particle`s? Could we do away with `Particle` altogether?

Reply:
The list of particles was for backwards compatibility. In particular many of the unit tests loop over lists of particles so this was a good sanity check. I guess now I have finished vectorising and am confident I've not changed the output we could get rid of Particle and edit all the unit tests to use the new class?

Reply:
So little on the fence either way with this one. The only thought I had, is there may be cases where you can't easily vectorise a function or operation, and therefore being able to loop over particles (as `Particle` states) to run a function/calculation, and use that list to repopulate the `Particles`, could be useful.

We probably should encourage people not to do this however, so should update our code/tests.